### ISSUE TITLE
Rollup reports circular dependency errors 

### ISSUE THREAD

**Issue by jmc420** [juliencrn/usehooks-ts#310]
Rollup is reporting these errors when bundling usehooks-ts

```
rollup v3.19.1
(!) Circular dependencies
node_modules/usehooks-ts/dist/esm/index.js -> node_modules/usehooks-ts/dist/esm/useClickAnyWhere/useClickAnyWhere.js -> node_modules/usehooks-ts/dist/esm/index.js
node_modules/usehooks-ts/dist/esm/index.js -> node_modules/usehooks-ts/dist/esm/useCountdown/useCountdown.js -> node_modules/usehooks-ts/dist/esm/index.js
node_modules/usehooks-ts/dist/esm/index.js -> node_modules/usehooks-ts/dist/esm/useDarkMode/useDarkMode.js -> node_modules/usehooks-ts/dist/esm/index.js
...and 16 more
(!) "this" has been rewritten to "undefined"
https://rollupjs.org/troubleshooting/#error-this-is-undefined
node_modules/usehooks-ts/dist/esm/useCopyToClipboard/useCopyToClipboard.js
1: var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
                    ^
2:     function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
3:     return new (P || (P = Promise))(function (resolve, reject) {
...and 1 other occurrence
node_modules/usehooks-ts/dist/esm/useFetch/useFetch.js
1: var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
                    ^
2:     function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
3:     return new (P || (P = Promise))(function (resolve, reject) {
...and 1 other occurrence

```

**Comment by PSoltes** [comment]
seconding this
![circulardeps](https://github.com/juliencrn/usehooks-ts/assets/22454458/777ff4f7-9398-4cca-8c31-70ab05dc3abb)

**Comment by juliencrn** [comment]
Please could you provide a minimal reproduction of your problem?

**Comment by juliencrn** [comment]
Hi all,

This error message reveals two different issues

The first part is due to the circular dependencies indeed, and it should be fixed by #440

```
rollup v3.19.1
(!) Circular dependencies
node_modules/usehooks-ts/dist/esm/index.js -> node_modules/usehooks-ts/dist/esm/useClickAnyWhere/useClickAnyWhere.js -> node_modules/usehooks-ts/dist/esm/index.js
node_modules/usehooks-ts/dist/esm/index.js -> node_modules/usehooks-ts/dist/esm/useCountdown/useCountdown.js -> node_modules/usehooks-ts/dist/esm/index.js
node_modules/usehooks-ts/dist/esm/index.js -> node_modules/usehooks-ts/dist/esm/useDarkMode/useDarkMode.js -> node_modules/usehooks-ts/dist/esm/index.js
...and 16 more
```

---

The next part is due to an incorrect ESM setup. It can't be fixed right now without introducing breaking changes but will be for sure in the incoming v3, ESM-first major release.

```
(!) "this" has been rewritten to "undefined"
https://rollupjs.org/troubleshooting/#error-this-is-undefined
node_modules/usehooks-ts/dist/esm/useCopyToClipboard/useCopyToClipboard.js
1: var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
                    ^
2:     function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
3:     return new (P || (P = Promise))(function (resolve, reject) {
...and 1 other occurrence
node_modules/usehooks-ts/dist/esm/useFetch/useFetch.js
1: var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
                    ^
2:     function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
3:     return new (P || (P = Promise))(function (resolve, reject) {
...and 1 other occurrence
```
